Upstream-base: efa6fc84680e48bfdca7e6ac245d8abf333ae617

--- a/src/MSL/wprintf.c
+++ b/src/MSL/wprintf.c
@@ -1141,7 +1141,11 @@
                 buff_ptr = va_arg(arg, wchar_t*);
 
                 if (buff_ptr == 0) {
+#if defined(MSCHARGED_NATIVE) && defined(__cplusplus)
+                    buff_ptr = const_cast<wchar_t*>(L"");
+#else
                     buff_ptr = L"";
+#endif
                 }
 
                 if (format.alternate_form) {
@@ -1160,7 +1164,11 @@
             } else {
                 strptr = va_arg(arg, char*);
                 if (strptr == 0) {
+#if defined(MSCHARGED_NATIVE) && defined(__cplusplus)
+                    strptr = const_cast<char*>("");
+#else
                     strptr = "";
+#endif
                 }
 
                 if (format.alternate_form) {
